File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/V086UserEventHandler.java
package org.emulinker.kaillera.controller.v086.action;

import org.emulinker.kaillera.controller.v086.V086ClientHandler;
import org.emulinker.kaillera.model.event.UserEvent;

public interface V086UserEventHandler<T extends UserEvent> {
  @Override
  public String toString();

  public void handleEvent(T event, V086ClientHandler clientHandler);

  public int getHandledEventCount();
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/V086UserEventHandler.kt
package org.emulinker.kaillera.controller.v086.action

import org.emulinker.kaillera.controller.v086.V086ClientHandler
import org.emulinker.kaillera.model.event.UserEvent

interface V086UserEventHandler<in T : UserEvent?> {
  @Deprecated("Structure this in a different way")
  override fun toString(): String

  fun handleEvent(event: T, clientHandler: V086ClientHandler)
  val handledEventCount: Int
}
